mlflow/models/evaluation/default_evaluator.py

```diff
@@ -203,33 +231,39 @@ def _get_common_classifier_metrics(
             sample_weight=sample_weights,
         ),
     }
+
     if y_proba is not None:
-        metrics["log_loss"] = sk_metrics.log_loss(
-            y_true, y_proba, labels=labels, sample_weight=sample_weights
+        metrics["log_loss"] = _calculate_log_loss(
+            y_true, y_proba, labels=labels, sample_weights=sample_weights
         )
-
     return metrics
 
 
 def _get_binary_classifier_metrics(
     *, y_true, y_pred, y_proba=None, labels=None, pos_label=1, sample_weights=None
 ):
-    tn, fp, fn, tp = sk_metrics.confusion_matrix(y_true, y_pred).ravel()
-    return {
-        "true_negatives": tn,
-        "false_positives": fp,
-        "false_negatives": fn,
-        "true_positives": tp,
-        **_get_common_classifier_metrics(
-            y_true=y_true,
-            y_pred=y_pred,
-            y_proba=y_proba,
-            labels=labels,
-            average="binary",
-            pos_label=pos_label,
-            sample_weights=sample_weights,
-        ),
-    }
+    try:
+        tn, fp, fn, tp = sk_metrics.confusion_matrix(y_true, y_pred).ravel()
+        return {
+            "true_negatives": tn,
+            "false_positives": fp,
+            "false_negatives": fn,
+            "true_positives": tp,
+            **_get_common_classifier_metrics(
+                y_true=y_true,
+                y_pred=y_pred,
+                y_proba=y_proba,
+                labels=labels,
+                average="binary",
+                pos_label=pos_label,
+                sample_weights=sample_weights,
+            ),
+        }
+    except ValueError:
```

Comment:
Is there particular error message we can include in the catch condition?

Reply:
I added some description in the context manager for what we're intending to do here - we're actually trying to fairly broadly catch and eat these exceptions if that private env var is set so that hyperparameter iterations in automl won't keep failing for smaller data sets / class imbalance issues. The word from training team was that customers don't care about these metrics for the tuning iteration runs - only the final run, which is carefully split (at the cost of runtime & memory pressure) for getting the final evaluation scores of the best model.